### PR TITLE
DOC: Fixed outdated doc-string, added missing default values, added missing optional parameters for some io classes

### PR DESCRIPTION
**Pull request by terrytangyuan**
Some doc-strings are outdated, in a sense that there are missing newly added optional parameters and default values.

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
@terrytangyuan The changes look good, but, the `data.py` is now maintained at https://github.com/pydata/pandas-datareader. So maybe you can do a PR for that file there

File: pandas/io/data.py [hunk 1]
```diff
@@ -53,12 +53,17 @@ def DataReader(name, data_source=None, start=None, end=None,
...
         the data source ("yahoo", "google", "fred", or "ff")
-    start : {datetime, None}
+    start : datetime, default: None
         left boundary for range (defaults to 1/1/2010)
-    end : {datetime, None}
+    end : datetime, default: None
         right boundary for range (defaults to today)
+    retry_count : int, default 3
+        Number of times to retry query request.
+    pause : int, default 0.001
```
Comment (jorisvandenbossche) [inline pandas/io/data.py]: this should be float I assume? (or 'numeric')

**Comment by terrytangyuan** [comment]
@jorisvandenbossche Nice catch! Thanks for checking. I've made the changes. I'll submit another PR to make changes on the new file.

**Comment by terrytangyuan** [comment]
@jorisvandenbossche I believe I only made small changes. No idea why Travis failed. Could you help me check?

**Comment by terrytangyuan** [comment]
@jorisvandenbossche On green now. Can you merge?

File: pandas/io/data.py [hunk 2]
```diff
@@ -400,26 +405,26 @@ def get_data_yahoo(symbols=None, start=None, end=None, retry_count=3,
     ----------
     symbols : string, array-like object (list, tuple, Series), or DataFrame
         Single stock symbol (ticker), array-like object of symbols or
-        DataFrame with index containing stock symbols.
+        DataFrame with index containing stock symbols, default: None
```
Comment (kawochen) [inline pandas/io/data.py]: Should be two lines above?

Comment (terrytangyuan) [inline pandas/io/data.py]: @kawochen Do you mean the `default: None`?

Comment (kawochen) [inline pandas/io/data.py]: Yes.  On line 406 instead (not sure but it looks like that's how other function parameters are documented)?

**Comment by terrytangyuan** [comment]
Thanks @kawochen Fixed and on green now.

**Comment by jreback** [comment]
I guess this is fine. @terrytangyuan prob strip off the stuff for ``pandas-datareader`` and submit there as well.

**Pull request merged**